### PR TITLE
aws ec2 metadata module

### PR DESCRIPTION
**Pull request by depombo**

### CODE REVIEW

File: src/modules/0.0.6/aws_ec2_metadata/entity/instance_metadata.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    onDelete: 'CASCADE',
+  })
+  @JoinColumn({
+    name: 'id'
+  })
+  id: number;
+
+  @Column()
+  @cloudId
+  instanceId: string;
```
Comment (aguillenv) [inline src/modules/0.0.6/aws_ec2_metadata/entity/instance_metadata.ts:34]: hmmm but why aren't you joining by the `instanceId`? It is only possible to have values on this table if an instance already exists.

Comment (depombo) [reply src/modules/0.0.6/aws_ec2_metadata/entity/instance_metadata.ts]: discussed offline. `instanceId` can be nullable in the `instance` table which violates the fk constraint

File: test/modules/aws-security-group-integration.ts [hunk 2]
```diff
@@ -304,7 +304,7 @@ describe('Security Group install/uninstall', () => {
     true).then(...finish(done)));
 
   it('uninstalls the Security Group module', uninstall(
-    ['aws_rds', 'aws_ecs_fargate', 'aws_elb', 'aws_security_group', 'aws_ec2'],
+    ['aws_rds', 'aws_ecs_fargate', 'aws_elb', 'aws_security_group', 'aws_ec2', 'aws_ec2_metadata'],
```
Comment (dfellis) [inline test/modules/aws-security-group-integration.ts:307]: Is this change necessary?

File: test/modules/aws-vpc-integration.ts [hunk 3]
```diff
@@ -131,7 +131,7 @@ describe('VPC install/uninstall', () => {
     true).then(...finish(done)));
 
   it('uninstalls the VPC module', uninstall(
-    ['aws_vpc', 'aws_ecs_fargate', 'aws_security_group', 'aws_rds', 'aws_elb', 'aws_ec2']));
+    ['aws_vpc', 'aws_ecs_fargate', 'aws_security_group', 'aws_rds', 'aws_elb', 'aws_ec2', 'aws_ec2_metadata']));
```
Comment (dfellis) [inline test/modules/aws-vpc-integration.ts:134]: Same here. This shouldn't be needed

**Review by dfellis — APPROVED** [review]
Two small comments. Would like them reverted as the EC2 module should continue to work without the metadata module

**Review by aguillenv — APPROVED** [review]
